tests/CSharp/CSharp.h

```diff
@@ -1484,11 +1484,7 @@ struct TestVariableWithoutType
     static constexpr auto variable = create(n...);
 };
 
-class DLL_API StringMarshall
-{
-public:
-    static const char* TestCSharpString(const char* in, CS_OUT const char** out);
-    static const wchar_t* TestCSharpStringWide(const wchar_t* in, CS_OUT const wchar_t** out);
-    static const char16_t* TestCSharpString16(const char16_t* in, CS_OUT const char16_t** out);    
-    static const char32_t* TestCSharpString32(const char32_t* in, CS_OUT const char32_t** out);
-};
+DLL_API const char* TestCSharpString(const char* in, CS_OUT const char** out);
```

Comment:
Exactly the kind of simplification I meant, thank you. I think that you no longer need the out parameters so please remove them and I'll merge.

Reply:
I can do that but note that when I added those output parameters some issues appeared like https://github.com/mono/CppSharp/blob/7055068107057e0bfb4e8abf62dac78fedb1f8c5/src/Generator/Types/Std/Stdlib.cs#L270 and https://github.com/mono/CppSharp/blob/7055068107057e0bfb4e8abf62dac78fedb1f8c5/src/Generator/Types/Std/Stdlib.cs#L277 since __retPtr would be declared twice (one for the output parameter and one for the return)

Reply:
So they help you test syntax too. OK, let them stay.